### ISSUE TITLE
LangSmith "Criteria()" and "LabeledCriteria()" do not accept custom criteria in TypeScript, despite examples

### ISSUE THREAD

**Issue by w-biggs** [langchain-ai/langchainjs#5188]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

Neither of these work. The first is what is implied to work by the comments in https://github.com/langchain-ai/langchainjs/blob/b9d86b16bd0788144f8f27cf5d18960c43da49af/langchain/src/smith/config.ts#L219, and the second is what I would expect to work based on the Python equivalent code.

```ts
const evalResult = await runOnDataset(
  llmFunction,
  datasetName,
  {
    evaluators: [{
      evaluatorType: "criteria",
      criteria: { "isCompliant": "Does the submission comply with the requirements of XYZ" }
    }],
  },
);
```

```ts
const compliantCriteria = Criteria({
  evaluatorType: "criteria",
  criteria: { "isCompliant": "Does the submission comply with the requirements of XYZ" }
});

const evalResult = await runOnDataset(
  llmFunction,
  datasetName,
  {
    evaluators: [compliantCriteria],
  },
);
```

### Description

There seems to be no way to use custom criteria in an criteria evaluator, which differs from the Python implementation.

The first method above does not seem to work -- the object is not compatible with what it wants to be in the evaluators array. The second method does not work either -- `Criteria()`'s argument is a `Criteria` type instead of a `CriteriaLike`, and if that line is `@ts-ignore`'d, it seems to fail with a 422 "expected a string" error when sending "feedback" to the LangSmith server.

Putting this issue in here because the code is in this repo and not in `LangSmith` proper (yet?)

### System Info

langchain@0.1.35
Node v21.5.0

**Comment by w-biggs** [comment]
@dosu This doesn't apply in this scenario -- I don't want to create an eval chain, I need to create an EvalConfig, since that is what `runOnDataset()` accepts as an argument.

**Comment by jacoblee93** [comment]
Sorry for the delay!

@dqbd could you have a look at this one?

**Comment by dqbd** [comment]
Hello @w-biggs!

The issue stems from the invalid type, which will be fixed in #5403. In the meantime, you can either:

1. Ignore the error with `@ts-expect-error` directive
```ts
{
  evaluators: [
    // @ts-expect-error Invalid type
    Criteria({
      isCompliant: "Does the submission comply with the requirements of XYZ",
    }),
  ],
}
```

2. pass `formatEvaluatorInputs` to obtain `prediction`, `input`, `reference`
